internal/reconciler/metricpipeline/status.go

```diff
@@ -53,90 +53,65 @@ func (r *Reconciler) updateStatus(ctx context.Context, pipelineName string, lock
 		return setCondition(ctx, r.Client, &pipeline, pending)
 	}
 
-	if err := r.updateGatewayStatus(ctx, pipelineName); err != nil {
-		return err
-	}
-
-	agentEnabled := isMetricAgentRequired(&pipeline)
+	return r.updateGatewayAndAgentStatus(ctx, pipeline)
 
-	if !agentEnabled {
-		return nil
-	}
-	return r.updateAgentStatus(ctx, pipelineName)
 }
 
-func (r *Reconciler) updateGatewayStatus(ctx context.Context, pipelineName string) error {
+func (r *Reconciler) updateGatewayAndAgentStatus(ctx context.Context, pipeline telemetryv1alpha1.MetricPipeline) error {
```

Comment:
So what would be the reason if the metric gteway and metric agent are both running?

Reply:
Reason would be MetricGatewayDeploymentReady in this case